rustual-boy-core/src/rom.rs

```diff
@@ -22,12 +22,15 @@ impl Rom {
         let mut vec = Vec::new();
         file.read_to_end(&mut vec)?;
 
-        let size = vec.len();
+        Rom::from_bytes(vec.into_boxed_slice())
+    }
+
+    pub fn from_bytes(mut bytes: Box<[u8]>) -> io::Result<Rom> {
```

Comment:
Note that taking a `&[u8]` would require copying the bytes rather than taking ownership, but that's not a problem here.